connector/profiler/src/main/scala/org/apache/spark/executor/profiler/ExecutorJVMProfiler.scala

```diff
@@ -0,0 +1,171 @@
+      profiler.execute(stopcmd)
+      logInfo("Code profiler stopped")
+      running = false
+      finishWriting()
+    }
+  }
+
+  private def startWriting(): Unit = {
+    if (profilerOutputDir.isDefined) {
+      val applicationId = conf.getAppId
+      val config = SparkHadoopUtil.get.newConfiguration(conf)
+      val appName = conf.get("spark.app.name");
+      val profilerOutputDirname = profilerOutputDir.get
+      val profileOutputFile =
+        s"$profilerOutputDirname/$applicationId/profile-$appName-exec-$executorId.jfr"
+      val fs = FileSystem.get(new URI(profileOutputFile), config);
+      val filenamePath = new Path(profileOutputFile)
+      outputStream = fs.create(filenamePath)
+      try {
+        if (fs.exists(filenamePath)) {
+          fs.delete(filenamePath, true)
+        }
+        logInfo(s"Copying executor profiling file to $profileOutputFile")
+        inputStream = new BufferedInputStream(new FileInputStream(s"$profilerLocalDir/profile.jfr"))
+        threadpool = ThreadUtils.newDaemonSingleThreadScheduledExecutor("profilerOutputThread")
+        threadpool.scheduleWithFixedDelay(new Runnable() {
+          override def run(): Unit = writeChunk()
+        }, writeInterval, writeInterval,
+          TimeUnit.SECONDS)
+        writing = true
+      } catch {
+        case e: Exception =>
+          logError("Failed to start code profiler", e)
+          if (threadpool != null) {
+            threadpool.shutdownNow()
+          }
+          if (inputStream != null) {
+            inputStream.close()
+          }
+          if (outputStream != null) {
+            outputStream.close()
+          }
+      }
+    }
+  }
+
+  private def writeChunk(): Unit = {
+    if (!writing) {
+      return
+    }
+    try {
+      // stop (pause) the profiler, dump the results and then resume. This is not ideal as we miss
+      // the events while the file is being dumped, but that is the only way to make sure that
+      // the chunk of data we are copying to dfs is in a consistent state.
+      profiler.execute(stopcmd)
+      profiler.execute(dumpcmd)
+      var remaining = inputStream.available()
+      profiler.execute(resumecmd)
+      while (remaining > 0) {
+        val read = inputStream.read(dataBuffer, 0, math.min(remaining, UPLOAD_SIZE))
+        outputStream.write(dataBuffer, 0, read)
+        remaining -= read
+      }
+    } catch {
+      case e: Exception => logError("Exception occurred while writing profiler output", e)
+    }
+  }
+
+  private def finishWriting(): Unit = {
+    if (profilerOutputDir.isDefined && writing) {
+      try {
+        // shutdown background writer
+        threadpool.shutdown()
+        threadpool.awaitTermination(30, TimeUnit.SECONDS)
+        // flush remaining data
+        writeChunk()
+        inputStream.close()
+        outputStream.close()
+      } catch {
+        case e: Exception =>
+          logError("Exception in completing profiler output", e)
```

Comment:
Could you add a descriptive message about what happens if we log and ignore this exception?